SG14/plf_colony.h

```diff
@@ -1,3867 +1,3867 @@
-
-					// Now we create a switch for the four different possible states (there are no additional instructions for state 0) depending on
-					// whether the left and right-hand skipfield nodes are non-zero (or out-of-bounds), by bit-shifting the right-hand test and
-					// logical-OR'ing it with the left-hand test:
-					switch (prev_skipfield | (after_skipfield << 1))
-					{
-						case 1: // previous erased consecutive elements, none following
-						{
-							*(new_location.skipfield_pointer - (value - 1)) = value - 1;
-							break;
-						}
-						case 2: // No previous consecutive erased points, at least one following ie. this was the prime erasure point
-						{
-							std::memmove(&*(new_location.skipfield_pointer + 2), &*(new_location.skipfield_pointer + 1), sizeof(skipfield_type) * (value - 2));
-							*(new_location.skipfield_pointer + 1) = value - 1;
-							break;
-						}
-						case 3: // both preceding and following consecutive erased elements
-						{
-							const skipfield_pointer_type start_node = new_location.skipfield_pointer - (value - 1);
-							const skipfield_type update_count = *start_node - value;
-							*start_node = value - 1;
-
-							std::memmove(&*(new_location.skipfield_pointer + 2), &*(start_node + 1), sizeof(skipfield_type) * (update_count - 1));
-							*(new_location.skipfield_pointer + 1) = update_count;
-						}
-					}
-
-					*new_location.skipfield_pointer = 0;
-					return new_location;
-				}
-			}
-		}
-		else // ie. newly-constructed colony, no insertions yet and no groups
-		{
-			initialize(pointer_allocator_pair.min_elements_per_group);
-
-			#ifdef PLF_COLONY_TYPE_TRAITS_SUPPORT
-				if (std::is_nothrow_copy_constructible<element_type>::value)
-				{
-					PLF_COLONY_CONSTRUCT(element_allocator_type, (*this), reinterpret_cast<pointer>(end_iterator.element_pointer++), element);
-				}
-				else
-			#endif
-			{
-				try
-				{
-					PLF_COLONY_CONSTRUCT(element_allocator_type, (*this), reinterpret_cast<pointer>(end_iterator.element_pointer++), element);
-				}
-				catch (...)
-				{
-					clear();
-					throw;
-				}
-			}
-
-			++end_iterator.skipfield_pointer;
-			total_number_of_elements = 1;
-			return begin_iterator;
-		}
-	}
-
-
-
-	#ifdef PLF_COLONY_MOVE_SEMANTICS_SUPPORT
-		iterator insert(element_type &&element) // The move-insert function is near-identical to the regular insert function, with the exception of the element construction method and is_nothrow tests.
-		{
-			if (end_iterator.element_pointer != NULL)
-			{
-				switch(((groups_with_erasures_list_head != NULL) << 1) | (end_iterator.element_pointer == reinterpret_cast<aligned_pointer_type>(end_iterator.group_pointer->skipfield)))
-				{
-					case 0:
-					{
-						const iterator return_iterator = end_iterator;
-
-						#ifdef PLF_COLONY_TYPE_TRAITS_SUPPORT
-							if (std::is_nothrow_move_const
```

Comment:
Line endings was changed (`CR LF` -> `LF`). Did this happen by an accident?

Reply:
Yeah, `git version 2.7.4 (Apple Git-66)` keeps wanting to change them to match the rest of the repo. [This trainwreck of a StackOverflow question](https://stackoverflow.com/questions/5834014/lf-will-be-replaced-by-crlf-in-git-what-is-that-and-is-it-important) suggests a lot of different fixes, but none of them worked on my system.

    warning: CRLF will be replaced by LF in SG14/plf_colony.h.
    The file will have its original line endings in your working directory.

...Aha! The culprit was that the SG14 repo had a hard-coded `.gitattributes` file checked in! That's not going to work if we've got some people on Windows and some people on Linux. I'm going to WP:BOLD and remove that file.  (It also had some cruft about how to diff .doc files and whatnot, but we don't have any .doc files in the SG14 repo.)  That'll stop git from trying to "fix" @mattreecebentley's line endings all the time.